app/src/main/java/com/tari/android/wallet/ui/activity/transactiondetails/TransactionDetailActivity.kt

```diff
@@ -0,0 +1,209 @@
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        overridePendingTransition(R.anim.enter_from_right, android.R.anim.fade_out)
+        emojiIdSummaryController = EmojiIdSummaryViewController(emojiSummaryView)
+        transaction = intent.getParcelableExtra(TRANSACTION_EXTRA_KEY)
+        if (transaction == null) finish()
+        setupUI()
+    }
+
+    @SuppressLint("SetTextI18n")
+    private fun setupUI() {
+        when (transaction) {
+            is CompletedTx -> {
+                when (transaction!!.direction) {
+                    Tx.Direction.INBOUND -> transactionTypeTv.text =
+                        getString(R.string.payment_received)
+                    Tx.Direction.OUTBOUND -> transactionTypeTv.text =
+                        getString(R.string.payment_sent)
+                }
+            }
+            is PendingInboundTx -> transactionTypeTv.text =
+                getString(R.string.pending_payment_received)
+            is PendingOutboundTx -> transactionTypeTv.text =
+                getString(R.string.pending_payment_sent)
+        }
+        transactionAmountTv.text = "%1$,.2f".format(transaction!!.amount.tariValue.toDouble())
+
+        val emojiId =
+            EmojiUtil.getEmojiIdForPublicKeyHexString(transaction!!.user.publicKeyHexString)
+        emojiIdSummaryController.display(emojiId)
+        transactionIdTv.text = "${getString(R.string.transaction_id)}:${transaction!!.id}"
+        if (transaction!!.message.isBlank()) {
+            noteLabel.visibility = View.INVISIBLE
+        }
+        transactionNoteTv.text = transaction!!.message
+        backBtn.setOnClickListener { onBackPressed() }
+        val user = transaction!!.user
+        if (user is Contact) {
+            contactContainer.visibility = View.VISIBLE
+            setAlias(user.alias)
+        } else {
+            addContactBtn.visibility = View.VISIBLE
+            contactContainer.visibility = View.GONE
+        }
+        if (transaction is CompletedTx) {
+            transactionFeeGroup.visibility = View.VISIBLE
+            transactionFeeTv.text = "${(transaction as CompletedTx).calculateFee()}"
+        } else {
+            transactionFeeGroup.visibility = View.GONE
+        }
+        createContactEt.setOnKeyListener{ _, keyCode, event ->
+            if (event.action == KeyEvent.ACTION_DOWN && keyCode == KeyEvent.KEYCODE_ENTER) {
+                val name = createContactEt.text?.toString()
+                if (name != null) {
+                    setAlias(name)
+                }
+                return@setOnKeyListener true
+            }
+            return@setOnKeyListener false
+        }
+
+        val calendar = Calendar.getInstance()
+        calendar.timeInMillis = transaction!!.timestamp.toLong() * 1000
+        formatDate(calendar)
+
+    }
+
+    private fun formatDate(cal: Calendar) {
```

Comment:
Did you try SimpleDateFormat or something like it? This function is redundant if you can format it with a formatter - please refer to other files for examples.